tests/common/plugins/pdu_controller/snmp_pdu_controllers.py

```diff
@@ -256,52 +280,48 @@ def get_outlet_status(self, outlet=None):
         Because of this, currently we just find out which PDU ports are connected to PSUs of which DUT. We cannot
         find out the exact mapping between PDU outlets and PSUs of DUT.
 
-        To overcome this limitation, the trick is to convert the specified outlet to integer, then calculate the mode
-        upon the number of PSUs on DUT. The calculated mode is used as an index to get PDU outlet ID stored in
-        self.pdu_ports. But still, we cannot guarantee that outlet 0 is first PSU of DUT, and so on.
-
         @param outlet: Optional. If specified, only return status of PDU outlet connected to specified PSU of DUT. If
                        omitted, return status of all PDU outlets connected to PSUs of DUT.
         @return: Return status of PDU outlets connected to PSUs of DUT in a list of dictionary. Example result:
-                     [{"outlet_id": 0, "outlet_on": True}, {"outlet_id": 1, "outlet_on": True}]
+                     [{"outlet_id": "0.0.1", "outlet_on": True}, {"outlet_id": "0.0.2", "outlet_on": True}]
                  The outlet in returned result is integer starts from 0.
         """
         results = []
         if not self.pduType:
-            logging.error('Unable to retrieve status: PDU type is unknown: pdu_ip {} dut {}'.format(self.controller, self.hostname))
+            logging.error('Unable to retrieve status: PDU type is unknown: pdu_ip {}'.format(self.controller))
             return results
 
+        if not outlet and not hostname:
+            # Return status of all outlets
+            ports = self.port_oid_dict.keys()
+        elif outlet:
+            ports = [ oid for oid in self.port_oid_dict.keys() if oid.endswith(outlet) ]
+            if not ports:
+                logging.error("Outlet ID {} doesn't belong to PDU {}".format(outlet, self.controller))
+        elif hostname:
+            hn = hostname.lower()
+            ports = [ self.port_label_dict[label]['port_oid'] for label in self.port_label_dict.keys() if hn in label ]
+            if not ports:
+                logging.error("{} device is not attached to any outlet of PDU {}".format(hn, self.controller))
+
```

Comment:
It's better to add a check here to avoid unnecessary operation and logging.
```
if not ports:
    return results
```

Reply:
As we discussed, I think the log is a good debug information. I'll keep the original structure.